gax/src/test/java/com/google/api/gax/rpc/internal/RoutingHeaderHelperTest.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google LLC nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.google.api.gax.rpc.internal;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.api.pathtemplate.PathTemplate;
+import com.google.common.collect.ImmutableMap;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class RoutingHeaderHelperTest {
+
+  @Test
+  public void shouldAddParamsIfThereAreMatchedValuesWithRoutingHeader() {
+    PathTemplate pathTemplate = PathTemplate.create("projects/**/{table_location=instances/*}");
+    ImmutableMap.Builder<String, String> params = ImmutableMap.builder();
+    String headerKey = "table_location";
+    RoutingHeaderHelper.addParams(
+        params, "projects/my_cozy_home/instances/living_room", headerKey, pathTemplate);
+    assertThat(params.build()).containsExactly(headerKey, "instances/living_room");
+  }
+
+  @Test
+  public void shouldNotAddParamsIfThereAreMatchedValuesWithNoRoutingHeaders() {
+    PathTemplate pathTemplate = PathTemplate.create("projects/**");
+    ImmutableMap.Builder<String, String> params = ImmutableMap.builder();
+    String headerKey = "table_location";
+    RoutingHeaderHelper.addParams(params, "projects/my_cozy_home/", headerKey, pathTemplate);
+    assertThat(params.build()).isEmpty();
```

Comment:
The 5 lines of each of these tests repeat. Can we factor it out into a private helper method 
One-liner would look something like this:
```java
assertThat(getRoutingHeaders("projects/**", "table_location", "projects/my_cozy_home/instances/living_room")).isEmpty();
```

Reply:
Extracted common code to a private method